homeassistant/components/simplisafe/__init__.py

```diff
@@ -351,6 +353,26 @@ async def set_system_properties(call: ServiceCall) -> None:
     ):
         async_register_admin_service(hass, DOMAIN, service, method, schema=schema)
 
+    current_options = {**entry.options}
+
+    async def async_reload_entry(_: HomeAssistant, updated_entry: ConfigEntry) -> None:
+        """Handle an options update.
+
+        This method will get called in two scenarios:
+          1. When SimpliSafeOptionsFlowHandler is initiated
+          2. When a new refresh token is saved to the config entry data
+
+        We only want #1 to trigger an actual reload.
+        """
+        nonlocal current_options
+        updated_options = {**updated_entry.options}
+
+        if updated_options == current_options:
+            return
+
+        current_options = updated_options
```

Comment:
You probably don't need this since the whole integration is about to be reloaded.

Reply:
But don't we need to know what the most current options are the next time around?

Reply:
If the integration reloads, won't it call unload, and then go through setup again?

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L969

Reply:
Ah, good point.